blaze-core/src/main/scala/org/http4s/blaze/Http1Stage.scala

```diff
@@ -234,22 +238,23 @@ object Http1Stage {
     else CachedEmptyBufferThunk
   }
 
-  /** Encodes the headers into the Writer, except the Transfer-Encoding header which may be returned
+  /** Encodes the headers into the Writer. Does not encode `Transfer-Encoding` or
+    * `Content-Length` headers, which are left for the body encoder. Adds
+    * `Date` header if one is missing and this is a server response.
+    *
     * Note: this method is very niche but useful for both server and client. */
-  def encodeHeaders(headers: Headers, rr: Writer, isServer: Boolean): Option[`Transfer-Encoding`] = {
-    var encoding: Option[`Transfer-Encoding`] = None
+  def encodeHeaders(headers: Iterable[Header], rr: Writer, isServer: Boolean): Unit = {
     var dateEncoded = false
-    headers.foreach { header =>
-      if (isServer && header.name == H.Date.name) dateEncoded = true
+    headers.view.filterNot(h => h.name == `Transfer-Encoding`.name ||
```

Comment:
As a foreword: I tend to prematurely optimize so bear with me.

I'm not sure this avoids creating a new collection; the stdlib is notorious for being a spaghetti bowl so its a whole lot of work to say for sure. Regardless it will make a bunch of new objects through `view`, `filterNot`, the necessary lambda, the lambda that deals with the `Not` part etc. Who knows how much will get inlined. This is just as easily dealt with using an if statement in the `foreach` and makes 0 new objects:
```scala
headers.foreach { h =>
// could change the predicate to: if (h.name != Tenc.name && h.name != Clen.name) ...
if (!(h.name == `Transfer-Encoding`.name || h.name == `Content-Length`.name)) {
  /* rest of the foreach */
}
```